python/tests/transfer_trigger_tests.py

```diff
@@ -0,0 +1,212 @@
+        after_update, before_delete, after_delete = create_triggers()
+
+        TM = TriggerManager()
+
+        TM.add_trigger(before_create)
+        self.assertTrue(Customer in TM.trigger_map and len(TM.trigger_map) == 1)
+        self.assertTrue("bc" in TM.trigger_map[Customer])
+        self.assertTrue(before_create in TM.trigger_map[Customer]["bc"])
+
+        TM.add_trigger(after_create)
+        self.assertTrue(Customer in TM.trigger_map and len(TM.trigger_map) == 1)
+        self.assertTrue("ac" in TM.trigger_map[Customer])
+        self.assertTrue(after_create in TM.trigger_map[Customer]["ac"])
+
+        TM.add_trigger(before_read)
+        self.assertTrue(Customer in TM.trigger_map and len(TM.trigger_map) == 1)
+        self.assertTrue("br" in TM.trigger_map[Customer])
+        self.assertTrue(before_read in TM.trigger_map[Customer]["br"])
+
+        TM.add_trigger(after_read)
+        self.assertTrue(Customer in TM.trigger_map and len(TM.trigger_map) == 1)
+        self.assertTrue("ar" in TM.trigger_map[Customer])
+        self.assertTrue(after_read in TM.trigger_map[Customer]["ar"])
+
+        TM.add_trigger(before_update)
+        self.assertTrue(Customer in TM.trigger_map and len(TM.trigger_map) == 1)
+        self.assertTrue("bu" in TM.trigger_map[Customer])
+        self.assertTrue(before_update in TM.trigger_map[Customer]["bu"])
+
+        TM.add_trigger(after_update)
+        self.assertTrue(Customer in TM.trigger_map and len(TM.trigger_map) == 1)
+        self.assertTrue("au" in TM.trigger_map[Customer])
+        self.assertTrue(after_update in TM.trigger_map[Customer]["au"])
+
+        TM.add_trigger(before_delete)
+        self.assertTrue(Customer in TM.trigger_map and len(TM.trigger_map) == 1)
+        self.assertTrue("bd" in TM.trigger_map[Customer])
+        self.assertTrue(before_delete in TM.trigger_map[Customer]["bd"])
+
+        TM.add_trigger(after_delete)
+        self.assertTrue(Customer in TM.trigger_map and len(TM.trigger_map) == 1)
+        self.assertTrue("ad" in TM.trigger_map[Customer])
+        self.assertTrue(after_delete in TM.trigger_map[Customer]["ad"])
+
+    def test_trigger_manager_multiple_add(self):
+        """ Test if adding multiple triggers using .add_triggers works for 
+            all kinds of triggers """
+
+        TM = TriggerManager()
+
+        for t_kind in trigger_kinds():
+
+            TM.add_triggers(list(create_triggers()))
+            # Check if the pcc_type was addded and that their is only one of them
+            self.assertTrue(Customer in TM.trigger_map and len(TM.trigger_map) == 1)
+            # Check that this kind of trigger is associated with the pcc_type
+            self.assertTrue(t_kind in TM.trigger_map[Customer])
+            # Check that this kind of trigger has a TriggerProcedure in it's list
+            self.assertTrue(TriggerProcedure == type(TM.trigger_map[Customer][t_kind][0]))
```

Comment:
assertEqual, or use isinstance(TM......, TriggerProcedure)
never use 
cls == type(obj) for a check anywhere. It's better to use isinstance(obj, type)